modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -974,11 +975,19 @@ impl Wallet {
 
             // TODO: use batching for mainnet syncing
             trace!(block = height, "Fetching block hash");
-            let block_hash = self
-                .btc_rpc
-                .get_block_hash(height as u64)
+            let block_hash =
+                retry(
```

Comment:
Out-of-scope for this pr: should we consider making all calls to IBitcoindRpc retrying internally?